Tests/Unit/Microsoft365DSC/Microsoft365DSC.EXORoleGroup.Tests.ps1

```diff
@@ -0,0 +1,175 @@
+[CmdletBinding()]
+param(
+)
+$M365DSCTestFolder = Join-Path -Path $PSScriptRoot `
+    -ChildPath '..\..\Unit' `
+    -Resolve
+$CmdletModule = (Join-Path -Path $M365DSCTestFolder `
+        -ChildPath '\Stubs\Microsoft365.psm1' `
+        -Resolve)
+$GenericStubPath = (Join-Path -Path $M365DSCTestFolder `
+        -ChildPath '\Stubs\Generic.psm1' `
+        -Resolve)
+Import-Module -Name (Join-Path -Path $M365DSCTestFolder `
+        -ChildPath '\UnitTestHelper.psm1' `
+        -Resolve)
+
+$Global:DscHelper = New-M365DscUnitTestHelper -StubModule $CmdletModule `
+    -DscResource 'EXORoleGroup' -GenericStubModule $GenericStubPath
+Describe -Name $Global:DscHelper.DescribeHeader -Fixture {
+    InModuleScope -ModuleName $Global:DscHelper.ModuleName -ScriptBlock {
+        Invoke-Command -ScriptBlock $Global:DscHelper.InitializeScript -NoNewScope
+
+        BeforeAll {
+            $secpasswd = ConvertTo-SecureString 'test@password1' -AsPlainText -Force
+            $Credential = New-Object System.Management.Automation.PSCredential ('tenantadmin@mydomain.com', $secpasswd)
+
+            $Global:PartialExportFileName = 'c:\TestPath'
+
+            Mock -CommandName Save-M365DSCPartialExport -MockWith {
+            }
+
+            Mock -CommandName New-M365DSCConnection -MockWith {
+                return 'Credentials'
+            }
+
+            Mock -CommandName Get-PSSession -MockWith {
+            }
+
+            Mock -CommandName Remove-PSSession -MockWith {
+            }
+
+            # Mock Write-Host to hide output during the tests
+            Mock -CommandName Write-Host -MockWith {
+            }
+            Function get-RoleGroup{
```

Comment:
Shy are you adding these functions here?

My guess is that you want to fake these cmdlets, but those so called "stubs" should be placed in this file:
https://github.com/microsoft/Microsoft365DSC/blob/Dev/Tests/Unit/Stubs/Microsoft365.psm1

That way you can Mock them later in the tests.

Reply:
added the Functions as stubs